### PR TITLE
[Tests-Only] Separated similar scenarios webUISharingInternalUsers1/2

### LINKED ISSUE #36890: Refactor webUISharingInternalUsers1/2 suite features
## Description
Being a large suite, `webUISharingInternalUsers` suite is splitted #36888 
During this, feature files with random (not descriptive) names are created

## TODO
- [x] Separate similar scenarios and glue them up inside a feature with reasonable name #36899

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Similar feature scenarios inside webUISharingInternalUsers1/2 suite are separated into reasonable features files

## Related Issue

- Fixes #36890

## Motivation and Context
see #36890

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/webUISharingInternalUsers2/shareWithUsers2.feature [hunk 1]
```diff
@@ -1,282 +0,0 @@
...
-    And user "user0" has created folder "/simple-folder"
-    And user "user0" has logged in using the webUI
-    And user "user0" has shared folder "simple-folder" with user "user1"
-    And the user has opened the share dialog for folder "simple-folder"
-    When the user sends the share notification by email using the webUI
-    Then a notification should be displayed on the webUI with the text "Email notification was sent!"
-    And the email address "user1@example.org" should have received an email with the body containing
-      """
-      just letting you know that user0 shared simple-folder with you.
-      """
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers2/shareWithUsers2.feature]: where did this scenario go?

Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers2/shareWithUsers2.feature]: inside miscShareWithUsers.feature:276

File: tests/acceptance/features/webUISharingInternalUsers1/miscShareWithUsers.feature [hunk 2]
```diff
@@ -328,210 +272,23 @@ Feature: Sharing files and folders with internal users
     When the user opens the share dialog for file "lorem.txt"
     Then the user should not be able to send the share notification by email using the webUI
 
-  Scenario: user shares a file with another user with uppercase username
-    Given user "user1" has been created with default attributes and skeleton files
+  @mailhog @skipOnOcV10.3
+  Scenario: user without email should be able to send notification by email when allow share mail notification has been enabled
```
Comment (kiranparajuli589) [inline tests/acceptance/features/webUISharingInternalUsers1/miscShareWithUsers.feature]: @phil-davis

Comment (phil-davis) [reply tests/acceptance/features/webUISharingInternalUsers1/miscShareWithUsers.feature]: OK, good stufff. The diff listing was just a bit difficult.

File: tests/acceptance/features/webUISharingInternalUsers2/shareWithUsers2.feature (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/webUISharingInternalUsers2/shareWithUsers2.feature]: see other comments - it is there

**Review by phil-davis — APPROVED** [review]

**Pull request merged**